promcda/utils/utils_for_main.py

```diff
@@ -529,6 +230,29 @@ def save_dict(dictionary: dict, folder_path: str, filename: str):
         logging.error(f"Error while dumping the dictionary into a pickle file: {e}")
 
 
+def preprocess_enums(data) -> Union[Union[dict, list[str]], Any]:
```

Comment:
Return type is unnecessarily complex and confusing. It can be simplified to just `Any` instead of showing that it can return `dict` or `list` or `any`. 
`Any` appears to be superset of all types. Also your return documentation in this method is ambiguous.